core/src/main/java/com/amplifyframework/core/DeveloperMenuActivity.java

```diff
@@ -29,27 +27,34 @@
  * This is the activity to display the developer menu.
  */
 public final class DeveloperMenuActivity extends FragmentActivity {
+
+    // Indicates whether this activity has started.
+    private static boolean activityStarted;
     // Parent layout for the developer menu.
     private View devMenuLayout;
     // Detect and handle shake events.
     private ShakeDetector detector;
-    // Manages app navigation.
-    private NavController navController;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_dev_menu);
         devMenuLayout = findViewById(R.id.dev_layout);
         devMenuLayout.setFocusable(true);
-        devMenuLayout.setVisibility(View.GONE);
-        detector = new ShakeDetector(getApplicationContext(), this::changeVisibility);
+        detector = new ShakeDetector(getApplicationContext(), this::finish);
```

Comment:
I believe she supports both which I think is the best customer experience.

Reply:
I was a little concerned that if a customer does a particularly strong shake, it could be recognized as 2 shakes, which would show and then immediately hide the menu, which I don't think would be an expected experience.   I don't feel too strongly though, I'm good to try it with this functionality.

Reply:
I added a duration threshold in ShakeDetector that only calls the given callback method if a minimum amount of time has passed since the shake started, in order to prevent the developer menu from appearing and then immediately disappearing if there's a strong shake.